File: frontend/src/app/core/services/direccion.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { ApiResponse, PagedResponse } from '../models/api-response.model';
import { Direccion, DireccionRequest } from '../models/direccion.model';

@Injectable({ providedIn: 'root' })
export class DireccionService {
  constructor(private api: ApiService) {}

  getById(id: number): Observable<ApiResponse<Direccion>> {
    return this.api.get(`/direcciones/${id}`);
  }

  create(data: DireccionRequest): Observable<ApiResponse<Direccion>> {
    return this.api.post('/direcciones', data);
  }

  update(id: number, data: DireccionRequest): Observable<ApiResponse<Direccion>> {
    return this.api.put(`/direcciones/${id}`, data);
  }

  delete(id: number): Observable<ApiResponse<void>> {
    return this.api.delete(`/direcciones/${id}`);
  }
}

File: backend/src/main/java/bo/capital/tec/pet/service/DireccionService.java
package bo.capital.tec.pet.service;

import bo.capital.tec.pet.common.response.PagedResponse;
import bo.capital.tec.pet.dto.DireccionRequestDTO;
import bo.capital.tec.pet.dto.DireccionResponseDTO;
import bo.capital.tec.pet.dto.DireccionSummaryDTO;

public interface DireccionService {
    DireccionResponseDTO create(DireccionRequestDTO dto);
    DireccionResponseDTO getById(Long id);
    PagedResponse<DireccionSummaryDTO> getAll(String calle, Long ciudadId, int page, int size);
    DireccionResponseDTO update(Long id, DireccionRequestDTO dto);
    void delete(Long id);
}
